### PR TITLE
Add 930s to confirmation page

### PR DESCRIPTION
**Pull request by leikkisa**
connects #11191

### CODE REVIEW

File: app/controllers/appeals_controller.rb [hunk 1]
```diff
@@ -114,8 +114,8 @@ def update
       set_flash_success_message
 
       render json: {
-        issuesBefore: request_issues_update.before_issues.map(&:ui_hash),
-        issuesAfter: request_issues_update.after_issues.map(&:ui_hash),
+        beforeIssues: request_issues_update.before_issues.map(&:ui_hash),
+        afterIssues: request_issues_update.after_issues.map(&:ui_hash),
```
Comment (leikkisa) [inline app/controllers/appeals_controller.rb]: I renamed these to make them more consistent with elsewhere because it was tripping me up a little

File: client/app/intake/pages/decisionReviewEditCompleted.jsx [hunk 2]
```diff
@@ -17,7 +18,7 @@ const leadMessageList = ({ veteran, formName, requestIssues, addedIssues }) => {
   const editMessage = () => {
     if (requestIssues.length === 0) {
       return 'removed';
-    } else if (_.every(addedIssues, (ri) => ri.withdrawalPending)) {
+    } else if (_.every(addedIssues, (ri) => ri.withdrawalPending || ri.withdrawalDate)) {
```
Comment (leikkisa) [inline client/app/intake/pages/decisionReviewEditCompleted.jsx]: This was a bug where it wouldn't consider a review withdrawn if a review had been partially withdrawn previously.  It still doesn't fully account for ineligible issues, but this is an improvement.

File: client/app/intake/reducers/appeal.js [hunk 3]
```diff
@@ -72,7 +73,7 @@ const updateFromServerIntake = (state, serverIntake) => {
     },
     veteranInvalidFields: {
       $set: {
-        veteranMissingFields: serverIntake.veteranInvalidFields.veteran_missing_fields.join(', '),
+        veteranMissingFields: _.join(serverIntake.veteranInvalidFields.veteran_missing_fields, ', '),
```
Comment (leikkisa) [inline client/app/intake/reducers/appeal.js]: This is fixing a bug that affects hot reload on the front end.

File: client/app/intake/pages/decisionReviewEditCompleted.jsx [hunk 4]
```diff
@@ -53,108 +54,64 @@ const leadMessageList = ({ veteran, formName, requestIssues, addedIssues }) => {
...
-      } else if (hasWithdrawnIssues) {
+      } else if (_.every(addedIssues, (ri) => ri.withdrawalPending || ri.withdrawalDate)) {
         return 'Review Withdrawn';
       }
 
       return 'Claim Issues Saved';
     };
 
+    const beforeEps = _.uniq(_.map(beforeIssues, 'endProductCode'));
+    const afterEps = _.uniq(_.map(afterIssues, 'endProductCode'));
```
Comment (leikkisa) [inline client/app/intake/pages/decisionReviewEditCompleted.jsx]: I think right now this still includes withdrawn issues, so need to fix that.

File: spec/feature/intake/end_product_correction_spec.rb [hunk 5]
```diff
@@ -45,12 +45,30 @@
     )
   end
 
+  let!(:test1) do
```
Comment (leikkisa) [inline spec/feature/intake/end_product_correction_spec.rb]: Oops, I'll take these out and write real tests. This was just for me.

File: app/models/request_issues_update.rb [hunk 6]
```diff
@@ -90,7 +90,11 @@ def after_issues
   end
 
   def edited_issues
-    @edited_issues ||= edited_request_issue_ids ? fetch_edited_issues : calculate_edited_issues
+    edited_request_issue_ids ? fetch_edited_issues : calculate_edited_issues
```
Comment (aroltsch) [inline app/models/request_issues_update.rb]: Why did you remove the memoization part?

File: app/controllers/claim_review_controller.rb [hunk 7]
```diff
@@ -60,14 +60,15 @@ def render_success
...
+                     afterIssues: request_issues_update.after_issues.map(&:ui_hash),
                      withdrawnIssues: request_issues_update.withdrawn_issues.map(&:ui_hash) }
     else
       render json: {
         redirect_to: nil,
-        issuesBefore: request_issues_update.before_issues.map(&:ui_hash),
-        issuesAfter: request_issues_update.after_issues.map(&:ui_hash),
+        beforeIssues: request_issues_update.before_issues.map(&:ui_hash),
+        afterIssues: request_issues_update.after_issues.map(&:ui_hash),
+        updatedIssues: request_issues_update.all_updated_issues.map(&:ui_hash),
```
Comment (aroltsch) [inline app/controllers/claim_review_controller.rb]: Confused why you need all issues instead of just sending back correction issues?

File: app/models/request_issues_update.rb [hunk 8]
```diff
@@ -191,7 +195,7 @@ def process_removed_issues!
 
   def correction
     @correction ||= RequestIssueCorrection.new(
-      review: review,
+      request_issues_update: self,
```
Comment (aroltsch) [inline app/models/request_issues_update.rb]: you could also pass `corrected_request_issue_ids` and `review` instead of passing the entire thing.

File: app/controllers/claim_review_controller.rb (hunk 7 above)
Comment (leikkisa) [reply app/controllers/claim_review_controller.rb]: On the confirmation page we need to know which EPs have changed, and what issues are on them.  "before" and "after" issues both include issues that haven't changed, and also don't include correction issues.  So updatedIssues includes the correction issues, removes "unchanged" issues, and also has their EP code data.

This is a refactor from what was there, due to a combination of there being bugs in the previous code :-/, and wanting to generalize the functionality to handle all of the EP types (before the confirmation page only grouped issues by rating/nonrating and didn't care about EP code).
